tools/walletextension/wallet_extension.go

```diff
@@ -224,14 +224,16 @@ func (we *WalletExtension) handleRequestWS(resp http.ResponseWriter, req *http.R
 	if err != nil {
 		return
 	}
-	fun(userConn)
+	// We handle requests in a loop until the connection is closed on the client side.
+	for !userConn.IsClosed() {
```

Comment:
This is the main change. We handle requests repeatedly until the user connection is closed.